src/GitHub.App/Services/TeamExplorerContext.cs

```diff
@@ -0,0 +1,221 @@
+
+                    if (newRepositoryPath != repositoryPath)
+                    {
+                        log.Information("Fire PropertyChanged event for ActiveRepository");
+                        ActiveRepository = repo != null ? service.CreateRepository(repo.RepositoryPath) : null;
+                        PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(nameof(ActiveRepository)));
+                    }
+                    else if (newBranchName != branchName)
+                    {
+                        log.Information("Fire StatusChanged event when BranchName changes for ActiveRepository");
+                        StatusChanged?.Invoke(this, EventArgs.Empty);
+                    }
+                    else if (newHeadSha != headSha)
+                    {
+                        log.Information("Fire StatusChanged event when HeadSha changes for ActiveRepository");
+                        StatusChanged?.Invoke(this, EventArgs.Empty);
+                    }
+                    else if (newTrackedSha != trackedSha)
+                    {
+                        log.Information("Fire StatusChanged event when TrackedSha changes for ActiveRepository");
+                        StatusChanged?.Invoke(this, EventArgs.Empty);
+                    }
+
+                    repositoryPath = newRepositoryPath;
+                    branchName = newBranchName;
+                    headSha = newHeadSha;
+                    solutionPath = newSolutionPath;
+                    trackedSha = newTrackedSha;
+                }
+            }
+            catch (Exception e)
+            {
+                log.Error(e, "Refreshing active repository");
+            }
+        }
+
+        static RepositoryInfo FindActiveRepository(object gitExt)
+        {
+            var activeRepositoriesProperty = gitExt.GetType().GetProperty("ActiveRepositories");
+            var activeRepositories = (IEnumerable<object>)activeRepositoriesProperty?.GetValue(gitExt);
+            var repo = activeRepositories?.FirstOrDefault();
+            if (repo == null)
+            {
+                return null;
+            }
+
+            var repositoryPathProperty = repo.GetType().GetProperty("RepositoryPath");
+            var repositoryPath = (string)repositoryPathProperty?.GetValue(repo);
+
+            var currentBranchProperty = repo.GetType().GetProperty("CurrentBranch");
+            var currentBranch = currentBranchProperty?.GetValue(repo);
+
+            var headShaProperty = currentBranch?.GetType().GetProperty("HeadSha");
+            var headSha = (string)headShaProperty?.GetValue(currentBranch);
+
+            var nameProperty = currentBranch?.GetType().GetProperty("Name");
+            var branchName = (string)nameProperty?.GetValue(currentBranch);
+
+            return new RepositoryInfo(repositoryPath, branchName, headSha);
+        }
+
+        class RepositoryInfo
```

Comment:
Don't we already have types that represent a repository somewhere that we can use here? Yet another representation of a repository feels very duplicated.